src/Template/Layout/js/libs/filters.js

```diff
@@ -1,14 +1,38 @@
 import Vue from 'vue';
+import { t } from 'ttag';
 
 /**
  * Converts a snake case string to title case.
  * Example: snake_case => Snake Case
  *
- * @param  {String} str the string to convert
+ * @param {String} str the string to convert
  * @return {String}
  */
-Vue.filter('humanize', function (str) {
-    return str.split('_').map(function (item) {
+Vue.filter('humanize', function(str) {
+    return str.split('_').map(function(item) {
         return item.charAt(0).toUpperCase() + item.substring(1);
     }).join(' ');
 });
+
+/**
+ * Capitalize a string.
+ *
+ * @param {String} str The string to capitalize
+ */
+Vue.filter('capitalize', function(str) {
+    if (!str) {
+        return '';
+    }
+
+    str = str.toString();
+    return str.charAt(0).toUpperCase() + str.slice(1);
+});
+
+/**
+ * Translate a string.
+ *
+ * @param {String} str The string to translate
+ */
+Vue.filter('translate', function(str) {
+    return t([str]);
+});
```

Comment:
vue extract cli won't find strings to translate, referred by `'string' | translate`.
not sure this is a good idea

Reply:
`ttag extract` wouldn't find that string anyway, if it were in a variable (like often happens with the object type). This filter should be used only for dynamic strings, maybe I could make it clearer in the method jsdoc?

Reply:
if you literally write in a template this, it's ok:
```
{{ __('an example) }}
```

if you write this, it's ko:
```
<: 'an_example' | translate :>
```
because, the component you added will always translate a string as a variable, even when it's not a dynamic variable. that's why I don't think it's a good idea

Reply:
I know.
In the second case you should do this (if you just can't use `__('an_example')` for some reason)
```
<: t`an_example` :>
```

and use the filter only in cases like this
```
<: object.type | translate :>
```
I thought it would be useful because it can be chained with other filters, like `<: object.type | capitalize | translate :>`

Reply:
why not using directly t?
```
<: t(object.type | capitalize) :>
```

Reply:
I tried but filters don't work inside a function call 😞 

If we want to avoid introducing this filter, we can solve the specific situation (for object types) by adding lowercase translation keys for all object types...

Reply:
I'd prefer `<: t(type) | capitalize :>` and translating lowercase strings too, and removing `translate` filter.